molgenis-metadata-manager/src/main/frontend/src/store/actions.js

```diff
@@ -170,6 +170,10 @@ export default {
    * Persist metadata changes to the database
    */
   [SAVE_EDITOR_ENTITY_TYPE] ({commit, state}: { commit: Function, state: State }) {
+    if (!state.editorEntityType.labelAttribute) {
+      commit(UPDATE_EDITOR_ENTITY_TYPE, {key: 'labelAttribute', value: state.editorEntityType.idAttribute})
```

Comment:
Changing the form input on save seems confusing to me, If a label is needed for entity to be valid, i suggest making it a required field ( both in back- and frontend ) and notifying the user about the need for a label attribute.